model/metadata/system_test.go

```diff
@@ -18,165 +18,181 @@
-				},
+			s: &System{Kubernetes: &Kubernetes{PodName: &podname}, DetectedHostname: &detected},
+		},
+		"k8s podUID": {
+			input: map[string]interface{}{
+				"kubernetes":        map[string]interface{}{"pod": map[string]interface{}{"uid": podUID}},
+				"detected_hostname": detected,
 			},
-			Output: common.MapStr{},
-		},
-		// namespace
-		{
-			System: System{
-				Hostname: &hostname,
-				Kubernetes: &Kubernetes{
-					Namespace: &podname, // any string
-				},
+			s: &System{Kubernetes: &Kubernetes{PodUID: &podUID}, DetectedHostname: &detected},
+		},
+		"k8s_namespace": {
+			input: map[string]interface{}{
+				"kubernetes":        map[string]interface{}{"namespace": namespace},
+				"detected_hostname": detected,
 			},
-			Output: common.MapStr{},
+			s: &System{Kubernetes: &Kubernetes{Namespace: &namespace}, DetectedHostname: &detected},
 		},
-		// non-nil kubernetes, currently not possible via intake
-		{
-			System: System{
-				Hostname:   &hostname,
-				Kubernetes: &Kubernetes{},
+		"k8s podname with configured hostname": {
+			input: map[string]interface{}{
+				"kubernetes":          map[string]interface{}{"pod": map[string]interface{}{"name": podname}},
+				"detected_hostname":   detected,
+				"configured_hostname": configured,
 			},
-			Output: common.MapStr{
-				"hostname": hostname,
-				"name":     hostname,
+			s: &System{Kubernetes: &Kubernetes{PodName: &podname}, DetectedHostname: &detected, ConfiguredHostname: &configured},
+		},
+		"k8s podUID with configured hostname": {
+			input: map[string]interface{}{
+				"kubernetes":          map[string]interface{}{"pod": map[string]interface{}{"uid": podUID}},
+				"detected_hostname":   detected,
+				"configured_hostname": configured,
 			},
+			s: &System{Kubernetes: &Kubernetes{PodUID: &podUID}, DetectedHostname: &detected, ConfiguredHostname: &configured},
 		},
-	}
-
-	for _, test := range tests {
-		output := test.System.fields()
-		assert.Equal(t, test.Output, output)
-	}
-}
-
-func TestSystemDecode(t *testing.T) {
-	host, name, arch, platform, ip := "host", "custom hostname", "amd", "osx", "127.0.0.1"
-	inpErr := errors.New("some error")
-	for _, test := range []struct {
-		input         interface{}
-		inputErr, err error
-		s             *System
-	}{
-		{input: nil, err: nil, s: nil},
-		{input: nil, inputErr: inpErr, err: inpErr, s: nil},
-		{input: "", err: errors.New("invalid type for system"), s: nil},
-		{
-			input: map[string]interface{}{"hostname": 1},
-			err:   utility.ErrFetch,
-			s:     &System{Hostname: nil, Architecture: nil, Platform: nil, IP: nil},
+		"k8s namespace with configured hostname": {
+			input: map[string]interface{}{
+				"kubernetes":          map[string]interface{}{"namespace": namespace},
+				"detected_hostname":   detected,
+				"configured_hostname": configured,
+			},
+			s: &System{Kubernetes: &Kubernetes{Namespace: &namespace}, DetectedHostname: &detected, ConfiguredHostname: &configured},
 		},
-		{
+		"k8s empty": {
```

Comment:
this one covers it currently but for the sake of regression, might be worth adding a case for just `detected_hostname` and `configured_hostname` set